Options Menu additions/fixes

forkymcforkface opened the pull request:
## Description
This PR adds the ability to change in the UI:
Hide Rom file extensions
Hide ROM Tags
~~Hides Dir '/' (no setting for it)!~~
~~Hide all saves folders~~

~~Moved:~~
~~Load progress bar to UI option~~

## Motivation and Context
Cleans up the UI for those that don't need all information

## How Has This Been Tested?
Tests were run on SC64 on Original N64. small change was made to load_rom.c to catch a crash. 


## Types of changes
- [X] Improvement (non-breaking change that adds a new feature)
- [ ] Bug fix (fixes an issue)
- [ ] Breaking change (breaking change)
- [ ] Documentation Improvement
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)

## Checklist:
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.


You agree with the license terms and that other license types may be granted with permission of the original `N64FlashcartMenu` project license holders.

Signed-off-by: 36553759+forkymcforkface@users.noreply.github.com


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Added settings to hide file extensions, ROM tags, and the saves folder in the file browser.
  - New options are available in the settings menu to toggle these visibility preferences.
  - The file browser and settings editor interfaces reflect these new customization options for improved user control.

- **Style**
  - Improved formatting and clarity in the settings editor display text.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

forkymcforkface committed:
options menu updated

added hide options for:
extensions
rom tags
save folder

moved show loading bar option to change in ui

load_rom.c modified to catch crash

forkymcforkface committed:
Update browser.c

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -11,6 +11,11 @@ static const char *format_switch (bool state) {
     }
 }
 
+static void set_loading_progress_bar_enabled_type (menu_t *menu, void *arg) {
```
This should not be available in the UI settings editor. It was added for a specific use case where users that required autoload and had scalers had issues with the flipping of resolution. The autoload function is now deprecated and due to be removed as we now have fast reboot.

forkymcforkface replied:
Understood I'll remove this portion of the code.

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -61,6 +83,12 @@ static void set_rumble_enabled_type (menu_t *menu, void *arg) {
 // }
 #endif
 
+static component_context_menu_t set_loading_progress_bar_enabled_context_menu = { .list = {
```
This should not be available in the UI settings editor. It was added for a specific use case where users that required autoload and had scalers had issues with the flipping of resolution. The autoload function is now deprecated and due to be removed as we now have fast reboot.

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -112,10 +158,14 @@ static component_context_menu_t set_rumble_enabled_type_context_menu = { .list =
 #endif
 
 static component_context_menu_t options_context_menu = { .list = {
+    { .text = "ROM Loading Bar", .submenu = &set_loading_progress_bar_enabled_context_menu },
```
This should not be available in the UI settings editor. It was added for a specific use case where users that required autoload and had scalers had issues with the flipping of resolution. The autoload function is now deprecated and due to be removed as we now have fast reboot.

networkfusion commented:
Given there is no real convention for Hide ROM Tags, I am not sure if it is a great idea (given your current implementation), so would require extensive testing.

networkfusion commented:
#203 #235

forkymcforkface commented:
> Given there is no real convention for ROM Tags, I am not sure if it is a great idea (given your current implementation), so would require extensive testing. Fully support the "hide save folders" and "hide ROM extensions" though.

I was looking at the code and it can be refactored into file_list.c which will be much cleaner if you are ok with reviewing those chsnges.

I implemented this code because Hiding ROM tags is a part of many emulation front end scrapers and metadata editors.

EmulationStation
RetroPie
Batocera
LaunchBox / BigBox
Pegasus
RGB-Pi UI

networkfusion commented:
In the latest develop, I have wrapped the autoload functionality in ifdef flags, so you can keep the progress bar changes, but keep them in the ifdef's...

forkymcforkface committed:
Merge branch 'Polprzewodnikowy:main' into hide_options

forkymcforkface committed:
Merge branch 'develop' into hide_options

forkymcforkface commented:
hmm moving it back into ifdef broke it. Im just going to remove it. No time to develop this further.

forkymcforkface commented:
ok, nvm figured it out.

forkymcforkface committed:
Update settings_editor.c

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -11,6 +11,11 @@ static const char *format_switch (bool state) {
     }
 }
 
+static void set_loading_progress_bar_enabled_type (menu_t *menu, void *arg) {
```
``
```suggestion
#ifdef ED64_AUTOLOAD_ROM
static void set_loading_progress_bar_enabled_type (menu_t *menu, void *arg) {
```

networkfusion replied:
```suggestion
#ifdef ED64_AUTOLOAD_ROM
static void set_loading_progress_bar_enabled_type (menu_t *menu, void *arg) {
```

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -11,6 +11,11 @@ static const char *format_switch (bool state) {
+static void set_loading_progress_bar_enabled_type (menu_t *menu, void *arg) {
+    menu->settings.loading_progress_bar_enabled = (bool)(uintptr_t)(arg);
+    settings_save(&menu->settings);
+}
```
```suggestion
}
#endif
```

networkfusion replied:
```suggestion
}
#endif
```

forkymcforkface committed:
Update src/menu/views/settings_editor.c

Co-authored-by: Robin Jones <networkfusion@users.noreply.github.com>

forkymcforkface committed:
Update src/menu/views/settings_editor.c

Co-authored-by: Robin Jones <networkfusion@users.noreply.github.com>

forkymcforkface committed:
Update src/menu/views/settings_editor.c

Co-authored-by: Robin Jones <networkfusion@users.noreply.github.com>

forkymcforkface committed:
Update src/menu/views/settings_editor.c

Co-authored-by: Robin Jones <networkfusion@users.noreply.github.com>

forkymcforkface commented:
Thank you @networkfusion for the review. I do not have a dev background so I kinda flail at this stuff.

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -158,16 +212,19 @@ static void draw (menu_t *menu, surface_t *d) {
 #ifdef ED64_AUTOLOAD_ROM
-        "  Autoload ROM      : %s\n"
-        "  ROM Loading Bar   : %s\n\n"
+        "  Autoload ROM      : %s\n\n"
```
Not sure this extra new line is needed... a screenshot to show it would help.

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -158,16 +212,19 @@ static void draw (menu_t *menu, surface_t *d) {
     ui_components_main_text_draw(
         ALIGN_LEFT, VALIGN_TOP,
         "\n\n"
-        "  Default Directory : %s\n\n"
```
Not sure this extra new line is needed... a screenshot to show it would help.

networkfusion commented:
> Thank you @networkfusion for the review. I do not have a dev background so I kinda flail at this stuff.

Quite alright.

Please fully retest your changes and feedback (adding screenshots and possibily videos), Hopefully we can merge this to the pre-release soon...

forkymcforkface committed:
Update settings_editor.c

forkymcforkface commented:
> > Thank you @networkfusion for the review. I do not have a dev background so I kinda flail at this stuff.
> 
> Quite alright.
> 
> Please fully retest your changes and feedback (adding screenshots and possibily videos), Hopefully we can merge this to the pre-release soon...

networkfusion commented:
> > > Thank you @networkfusion for the review. I do not have a dev background so I kinda flail at this stuff.
> > 
> > 
> > Quite alright.
> > Please fully retest your changes and feedback (adding screenshots and possibily videos), Hopefully we can merge this to the pre-release soon...
> 
> Will do, not trying to change anything that I did prior. It should all be the same. Its angry that we moved the loading bar out of that section.

I will try to take a quick look and probably push to your PR branch.

networkfusion committed:
Update settings_editor.c

Exclude set_loading_progress_bar_enabled_context_menu when ED64_AUTOLOAD_ROM is not enabled.

networkfusion committed:
Update settings_editor.c

Fix load bar setting option

networkfusion commented:
@forkymcforkface , building now.

forkymcforkface commented:
> @forkymcforkface , building now.

you beat me by 1min :) . I was just pushing the changes and it said no changes found.

forkymcforkface committed:
Update settings_editor.c

forkymcforkface committed:
Update settings_editor.c

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -163,11 +219,14 @@ static void draw (menu_t *menu, surface_t *d) {
-        "     Show Hidden Files : %s\n"
-        "     Use Saves folder  : %s\n"
-        "     Sound Effects     : %s\n"
+        "  To change the following menu settings, press 'A':\n\n"
```
```suggestion
        " To change the following menu settings, press 'A':\n"
```

forkymcforkface committed:
Update src/menu/views/settings_editor.c

This gap is supposed to be there. Allows for 
title
gap
option

Co-authored-by: Robin Jones <networkfusion@users.noreply.github.com>

forkymcforkface committed:
Update settings_editor.c

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -163,11 +219,14 @@ static void draw (menu_t *menu, surface_t *d) {
-        "     Show Hidden Files : %s\n"
-        "     Use Saves folder  : %s\n"
-        "     Sound Effects     : %s\n"
+        " To change the following menu settings, press 'A':\n"
```
```suggestion
        "To change the following menu settings, press 'A':\n"
```

networkfusion commented:
@coderabbit full review

networkfusion commented:
@forkymcforkface please dont take the code rabbit review as required, only pointers...

networkfusion commented on src/menu/views/settings_editor.c:
```diff
@@ -163,11 +219,14 @@ static void draw (menu_t *menu, surface_t *d) {
-        "     Show Hidden Files : %s\n"
-        "     Use Saves folder  : %s\n"
-        "     Sound Effects     : %s\n"
+        " To change the following menu settings, press 'A':\n\n"
```
```suggestion
        "To change the following menu settings, press 'A':\n\n"
```

forkymcforkface committed:
updated parent dir from / to ..

forkymcforkface committed:
Update src/menu/views/settings_editor.c

Co-authored-by: Robin Jones <networkfusion@users.noreply.github.com>

forkymcforkface committed:
refined hide saves folder option

networkfusion committed:
Merge branch 'develop' into hide_options

networkfusion committed:
Fix merge conflicts

Minor fix

networkfusion commented:
@forkymcforkface I have merged in the current develop branch and fixed the conflicts. As a note with I had to go with a better implementation of hiding the save folder, but thanks for the inspiration.

Also it is much better to create seperate PR's for each feature as much easier to review and feedback...

networkfusion committed:
Merge branch 'develop' into hide_options

networkfusion committed:
Fix merge

networkfusion committed:
Merge branch 'develop' into hide_options

networkfusion committed:
Merge branch 'develop' into hide_options

networkfusion committed:
Merge branch 'develop' into hide_options

rasky committed:
boot: reset value of FCR31 to the boot value (0) (#254)

This improves emulation of cold boot, as otherwise the FPU might start
in an unexpected state.

<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->

## Summary by CodeRabbit

* **Chores**
* Improved initialization procedures during system boot for enhanced
stability.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kaylagator committed:
Fix missing enum case for 1 mbit SRAM saves (#253)

## Description
This allows the menu to boot ROMs with 1 MBit saves.

## Motivation and Context
Without this I get "Error occured during save loading" when trying to
load a ROM using this save type in the header.

## How Has This Been Tested?
From a libdragon install, run `ed64romconfig --savetype sram1m file.z64`
on any ROM to change the header byte, then try to load it in the menu.
With the patch, the ROM should load successfully. Sample using a test
ROM here:

[test_sram1m.zip](https://github.com/user-attachments/files/21261604/test_sram1m.zip)

## Screenshots
None

## Types of changes
- [ ] Improvement (non-breaking change that adds a new feature)
- [X] Bug fix (fixes an issue)
- [ ] Breaking change (breaking change)
- [ ] Documentation Improvement
- [ ] Config and build (change in the configuration and build system,
has no impact on code or features)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

Signed-off-by: lategator <212063191+lategator@users.noreply.github.com>


<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->

## Summary by CodeRabbit

* **Bug Fixes**
* Improved compatibility for games using the 1Mbit SRAM save type,
ensuring proper save data handling.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Co-authored-by: Robin Jones <networkfusion@users.noreply.github.com>

networkfusion committed:
Create FUNDING.yml

networkfusion committed:
Merge branch 'develop' into hide_options

forkymcforkface committed:
Merge branch 'Polprzewodnikowy:main' into hide_options
